October 2024 Newsletter (#825)
RSE-Sheffield/RSE-Sheffield.github.io: _newsletters/2024-10-10-newsletter.md

- Weds 16th Oct - [CMI Town Hall](https://www.sheffield.ac.uk/machine-intelligence/events/shefai-community-meeting-8) Eighth CMI community and networking event.
- Open Access Week: Oct 21st-25th is International Open Access Week, and the university library is hosting a [programme of events](https://www.sheffield.ac.uk/library/open-access/open-access-week-2024) to explore the value and possibilities of open access scholarship.
    - Tues Oct 22nd 1-1.30pm - White Rose University Press - A bitesize introduction. [Book your place here](https://www.eventbrite.co.uk/e/white-rose-university-press-a-bitesize-introduction-tickets-1029731543767)
- Open Research Conversation's [Autumn-Winter Schedule](https://www.sheffield.ac.uk/openresearch/events/open-research-conversations) has now been announced with 5 Wednesday lunchtime events from October through February.
    - Weds Oct 23rd 12-1pm - Beyond open access: Democratising knowledge through accessible research outputs. [Book your place here](https://www.eventbrite.co.uk/e/beyond-open-access-democratising-knowledge-through-accessible-outputs-tickets-1028028971327).

#### Upcoming External Events

> - Until Nov 10th - [MATLAB Shorts Mini Hack](https://uk.mathworks.com/matlabcentral/contests/2024-matlab-mini-hack.html): Use upto 2,000 characters ofr MATLAB code to generated a MATLAB movie and win prizes.

Reviewer: ```suggestion
- Until Nov 10th - [MATLAB Shorts Mini Hack](https://uk.mathworks.com/matlabcentral/contests/2024-matlab-mini-hack.html): Use upto 2,000 characters of MATLAB code to generate a MATLAB movie and win prizes.
```

### Events

#### Upcoming Local Events

- SheffieldR User Group is hosting hybrid weekly [hackathons throughout October on Fridays](https://sheffieldr.github.io/posts/2024-10/) for those participating in HacktoberFest 2024.
- [CMI Town Hall](https://www.sheffield.ac.uk/machine-intelligence/events/shefai-community-meeting-8) (Weds 16th Oct) Eighth CMI community and networking event.
- Open Access Week: Oct 21st-25th is International Open Access Week, and the university library is hosting a [programme of events](https://www.sheffield.ac.uk/library/open-access/open-access-week-2024) to explore the value and possibilities of open access scholarship.
    - Tues Oct 22nd 1-1.30pm - White Rose University Press - A bitesize introduction. [Book your place here](https://www.eventbrite.co.uk/e/white-rose-university-press-a-bitesize-introduction-tickets-1029731543767)
- Open Research Conversation's [Autumn-Winter Schedule](https://www.sheffield.ac.uk/openresearch/events/open-research-conversations) has now been announced with 5 Wednesday lunchtime events from October through February.

>     - Weds Oct 23rd 12-1pm - Beyond open access: Democratising knowledge through accessible research outputs. [Book your place here](https://www.eventbrite.co.uk/e/beyond-open-access-democratising-knowledge-through-accessible-outputs-tickets-1028028971327).

Reviewer: Nice way of avoiding duplication between Open Access Week and Open Research Conversations :+1:

Author: I just put the next event from each, did notice the overlap.

#### Upcoming External Events

> - [pyOpenSci Fall Festival 2024](https://www.eventbrite.com/e/pyopensci-fall-festival-2024-registration-1000762657047) (Oct 29th - Nov 1st) - A week of events to empower scientists to write better code and share their outcomes openly.

Reviewer: @ptheywood has dropped this in late today, date will have passed come the next newsletter...

```suggestion
- [pyOpenSci Fall Festival 2024](https://www.eventbrite.com/e/pyopensci-fall-festival-2024-registration-1000762657047) (Oct 29th - Nov 1st) - A week of events to empower scientists to write better code and share their outcomes openly.
- [DiRAC GPU Hackathon, Durham University](https://dirac.ac.uk/training_events/dirac-nvidia-hackathon-2024/) (Nov 26-28th), application deadline 12pm October 16th. 
```

Adding here so its in chronological order.

> - [pyOpenSci Fall Festival 2024](https://www.eventbrite.com/e/pyopensci-fall-festival-2024-registration-1000762657047) (Oct 29th - Nov 1st) - A week of events to empower scientists to write better code and share their outcomes openly.

Reviewer: Other events have dates before link but realistically doesn't make much difference...

```suggestion
- Oct 29th to Nov 1st - [pyOpenSci Fall Festival 2024](https://www.eventbrite.com/e/pyopensci-fall-festival-2024-registration-1000762657047) - A week of events to empower scientists to write better code and share their outcomes openly.
```

> - [UK Technology Specialists Network Conference](https://itss.org.uk/uk-technology-specialists-network-conference-2025/) April 29-30th 2025

Reviewer: As above following dates first...

```suggestion
- April 29-30th 2025 - [UK Technology Specialists Network Conference](https://itss.org.uk/uk-technology-specialists-network-conference-2025/) 
```

- [Cambridge RSE Seminar series](https://rse.group.cam.ac.uk/seminars) Thursday's 1pm (currently scheduled till Dec 5th)
    - [Open source research software project life cycles: understanding project maturity successes, and challenge events](https://talks.cam.ac.uk/talk/index/219982) (Thurs Oct 17th 1pm)

### Articles, Blogs, Papers & Podcasts

#### Articles & Blogs

- [Dual licensing R packages with code and data](https://epiverse-trace.github.io/posts/data-licensing.html)
- [Technical Properties of Sustainable Software](https://bssw.io/blog_posts/technical-properties-of-sustainable-software)

#### Papers

> - [2024] [Research Software Engineering: Bridging Knowledge Gaps](https://drops.dagstuhl.de/entities/document/10.4230/DagRep.14.4.42)

Reviewer: As with the pre-print making it clear its not peer-reviewed pulication.

```suggestion
- [2024] **Seminar Report** [Research Software Engineering: Bridging Knowledge Gaps](https://drops.dagstuhl.de/entities/document/10.4230/DagRep.14.4.42)
```

- [2024] [Understanding and effectively mitigating code review anxiety](https://link.springer.com/article/10.1007/s10664-024-10550-9)

> - [2024] [A Vision on Open Science for the Evolution of Software Engineering Research and Practice](https://arxiv.org/abs/2405.12132)

Reviewer: I think its useful to mark pre-prints as such as they've not been peer-reviewed...

```suggestion
- [2024] **Pre-print** [A Vision on Open Science for the Evolution of Software Engineering Research and Practice](https://arxiv.org/abs/2405.12132)
```

Could easily go at the end of the line though, no particular style or preference here.

- [2024] [Just Enough Software Engineering for Domain Scientists in Research Software Development](https://ieeexplore.ieee.org/document/10603285)
- [2023] [The replication crisis has led to positive structural, procedural, and community changes](https://www.nature.com/articles/s44271-023-00003-2)

#### Podcasts

- Code for Thought - [Bonus: Green Computing at RSECon24](https://codeforthought.buzzsprout.com/1326658/episodes/15766448-en-bonus-green-computing-at-the-rse-conference-2024-in-newcastle)

### Training

#### At Sheffield

**FAIR<sup>2</sup>**

The [FAIR<sup>2</sup> for Research Software Training Programme](https://rse.shef.ac.uk/training/fair4rs/) has been developed
by the RSE, Data Analytics Service and Library teams and will be launched in October. An introduction to the programme

> will be given in a [LunchBytes session 2024-10-22](../events/lunchbytes-2024-10-22.html). The full syllabus is now live on [MyDevelopment](https://mydevelopment.csod.com/ui/lms-learning-details/app/curriculum/06573f55-12e4-44f1-be4c-fac5ea9d6851) (requires login).

Reviewer: `<pedantic>`Hard wrap to match rest of paragraph`</pedantic>`

```suggestion
will be given in a [LunchBytes session 2024-10-22](../events/lunchbytes-2024-10-22.html). The full syllabus is now live on 
[MyDevelopment](https://mydevelopment.csod.com/ui/lms-learning-details/app/curriculum/06573f55-12e4-44f1-be4c-fac5ea9d6851) (requires login).
```

Author: ![image](https://github.com/user-attachments/assets/79170afc-51f6-40bb-bdcf-4ac9c842c349)
It looks fine in my text editor without any changes 🤷‍♂️

Reviewer: My pedantry with line wraps is less to do with how it appears in editors which will do soft line-wraps, and in Emacs you get an indicator that these are soft-wraps with little arrows at the side...

![soft-wrapped-lines](https://github.com/user-attachments/assets/2b1ad4b6-d2de-46e3-8302-b1576f9fa64f)

...rather when it comes to line diffing and finding errors/rendering differences its much harder to see these in really long chunks of text. That is somewhat ameliorated with the `--word-diff=color` but I've just got a old habit of hard-wrapping lines (well not so much a habit, it just happens automatically in my editor or its done by linters).

As I say, its pedantic and doesn't matter as it doesn't affect the rendered content which is wrapped automatically based on the available space.

Author: Good diff checkers typically highlight affected characters within a line if tokens have changed but most of the line remains the same.

e.g.
![image](https://github.com/user-attachments/assets/f185b762-9143-4c52-a89e-2816e123a699)

or

![image](https://github.com/user-attachments/assets/c9cdf528-627e-4131-b8c1-9744e88b6c9b)

**Sheffield Bioinformatics Core**

> The Sheffield Bioinformatics Core is pleased to announce a schedule of courses for the new academic year. These courses provide an introduction to the R programming language, Unix environment (required for the use of the University's High Performance Computing facilities) and also specific applications in RNA-sequencing.

Reviewer: Pedantic line wrapping...

```suggestion
The Sheffield Bioinformatics Core is pleased to announce a schedule of courses for the new academic year. These courses 
provide an introduction to the R programming language, UNIX environment (required for the use of the University's High
 Performance Computing facilities) and also specific applications in RNA-sequencing.
```

Author: my text editor supports word-wrap, waste of screen space if it's force wrapped different to my editor window..

> You can now sign-up for these courses using myDevelopment (search for "bioinformatics"). Course charges still apply, and upon sign-up we will contact you for a valid grant code or cost centre. In-person attendance is encouraged, but online options are also available.

Reviewer: Pedantic line wrapping...

```suggestion
You can now sign-up for these courses using myDevelopment (search for "bioinformatics"). Course charges still apply, and 
upon sign-up we will contact you for a valid grant code or cost centre. In-person attendance is encouraged, but 
online options are also available.
```

- Data Manipulation and Visualization - October 28th, 30th and November 1st 13:00 - 16:00 - (£72 for students and £80 for staff)
- Introduction to RNA-seq analysis - November 12th 13:00 - 16:00 (£32 for students and £40 for staff)

> Introduction to the Command-Line for Bioinformatics - November 27th 13:00 - 16:00 (£32 for students and £80 for staff)

Reviewer: ```suggestion
- Introduction to the Command-Line for Bioinformatics - November 27th 13:00 - 16:00 (£32 for students and £80 for staff)
```

- Introduction to RNA-seq analysis in R - December 1st, 3rd and 5th 13:00 - 16:00 (£72 for students and £80 for staff)

Where multiple dates are listed, you will need to attend all dates. You can also find more information on these course on their website (https://sbc.shef.ac.uk)

#### Outside Sheffield

- [SSI Software Practices in the Social Sciences programme](https://www.software.ac.uk/blog/ssi-announces-research-software-practices-social-sciences-programme) - A new programme of 1-hour workshops covering topics such as Relational Databases, Reproducible Analysis and Data Cleanup for Social Scientists.
- [UK Data Service](https://ukdataservice.ac.uk/) provide access to a wide range of reusable social sciene data, and training to enable researchers to make the most of it.
    - Mon Oct 14th 10am - [Introduction to synthetic data for longitudinal data managers](https://ukdataservice.ac.uk/events/introduction-to-synthetic-data-for-longitudinal-data-managers/)

>     - Mon 10th 10am - [Safe Researcher Training](https://ukdataservice.ac.uk/events/safe-researcher-training-67/)

Reviewer: Link says its 14th Oct

```suggestion
    - Mon Oct 14th 10am - [Safe Researcher Training](https://ukdataservice.ac.uk/events/safe-researcher-training-67/)
```

    - [and more...](https://ukdataservice.ac.uk/training-events/)
    - with recordings of past trainings available on their [YouTube](https://www.youtube.com/ukdataservice)

### Funding Opportunities

- [FAIR-IMPACT's 3rd call](https://fair-impact.eu/3rd-open-call-route-2-support-opens-30-sept), deadline 4th Dec, hopes to fund applicants in four areas
    - Managing data types, schemas and vocabularies and crosswalks for FAIR researcher related metadata (in partnership with FAIRCORE4EOSC)
    - Creating EOSC compliant interoperability policies based on the EOSC Interoperability Framework (IF)
    - Testing the trustworthy and FAIR-enabling repositories prototype
    - Implementation of a shared API for semantic catalogues

> #### Jobs

Reviewer: `<pedantic>`Linting would suggest spacing`</pedantic>`

```suggestion

#### Jobs
```

> There are two positions within the University of Sheffield currently being advertised.

Reviewer: ```suggestion
```

Reviewer: Not currently positions being advertised and there was the announcement of a moratorium on recruitment.

Author: I think you're reviewing an old version this should be gone.

Reviewer: Indeed, apologies, I thought GitHub kept things fresh but it seems not.

> - 

Reviewer: ```suggestion
```

Check for advertised RSE and RSE-adjacent roles at the [RSE society's vacancies
board](https://society-rse.org/careers/vacancies/).

### Community

#### Digital Research Practice Support Community

The DRPS community is a group for people that support researchers in carrying out research in the digital age. Meetings
are held monthly, with discussions around events, training and opportunities related to the field.

You can join the google group
[here](https://groups.google.com/u/1/a/sheffield.ac.uk/g/digital-research-practice-support-community-group/about) to
stay informed.

> The next meeting is scheduled for 2pm on Wednesday 18th September 2024.

Reviewer: ```suggestion
The next meeting is scheduled for 2pm on Wednesday 6th November 2024.
```

Author: whoops

#### LunchBytes

[LunchBytes](https://rse.shef.ac.uk/community/lunch-bytes/) are short talks from the research community on research
software, data, and infrastructure.

##### LunchBytes needs YOU

LunchBytes are organised by and for the research software community at The University of Sheffield. If you'd like to
curate a session on a topic or present something, get in touch by emailing

> [lunchbytes-organisers-group@sheffield.ac.uk](mailto:lunchbytes-organisers-group@sheffield.ac.uk) or suggest topics

Reviewer: Was mentioned today that Farhad will be taking over organising these, we should make sure that he's on this mailing list.

I've tried joining [https://groups.google.com/a/sheffield.ac.uk/g/lunchbytes-organisers-group](https://groups.google.com/a/sheffield.ac.uk/g/lunchbytes-organisers-group) but without success, page isn't found.

Author: Yeah, I wasn't sure how to phrase that given nothing yet in the pipeline.

Not sure about the missing group, maybe wilby owned it so it got deleted?
